Dockerfile

```diff
@@ -88,9 +123,13 @@ ENV POETRY_VERSION="2.1.3"
 
 WORKDIR ${APP_PATH}
 
-RUN set -eux; \
-    python -m venv ${PHYLUM_VENV}; \
-    ${PHYLUM_VENV_PIP} install --no-cache-dir --upgrade pip setuptools
+RUN python -m venv ${PHYLUM_VENV}
+# Isolate the use of the `PHYLUM_TOKEN` secret to a single RUN command so as not
+# to leak the value in logs with `set -eux` command
+RUN --mount=type=secret,id=PHYLUM_TOKEN,env=PHYLUM_TOKEN,required=true \
+    ${PHYLUM_VENV_PIP} config set global.index-url \
+        "https://${PHYLUM_ORG}%2F${PHYLUM_GROUP}:${PHYLUM_TOKEN}@pypi.phylum.io/simple/"
```

Comment:
It would probably be easier to just pass the whole source URI as one secret, and then it would be possible to build an image without a Phylum account or direct network access to phylum.io.

Reply:
This sounds like a good idea. I'll work to make the change.

Reply:
Upon further reflection, I don't think this is a good change. The idea behind this PR is that the repo is fully protected by the Package Firewall and allowing for the creation of an image without a Phylum account, while helpful to some, introduces risk that our own use could be unprotected. This is really only a complication due to `phylum-ci` being a public repo...but @furi0us333 has previously directed that it shouldn't be treated any differently from our private repos and enforcing protection is acceptable here.